js_modules/dagit/packages/core/src/ui/BaseTag.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import styled from 'styled-components/macro';
+
+import {ColorsWIP} from './Colors';
+import {IconWrapper} from './Icon';
+
+interface Props {
+  fillColor?: string;
+  iconColor?: string;
+  textColor?: string;
+  icon?: React.ReactNode;
+  rightIcon?: React.ReactNode;
+  label?: React.ReactNode;
+}
+
+export const BaseTag = (props: Props) => {
+  const {
+    fillColor = ColorsWIP.Gray10,
+    iconColor = ColorsWIP.Gray900,
```

Comment:
I'm not a huge fan of putting iconColor on the Tag when the icon is a node and the color can also be passed on the IconWIP. It's not super clear whether I should do:

```
<Tag iconColor={ColorsWIP.Yellow500} icon={<IconWIP name="warning" />} />

OR 


<Tag icon={<IconWIP name="warning" color={ColorsWIP.Yellow500 />} />
```

I'd sort of prefer the latter or having the API change to something like

```
<Tag iconColor={ColorsWIP.Yellow500} icon="warning" />
```

Reply:
Yeah, I could be in favor of limiting the available icons on `Tag`. It certainly simplifies the existing `intent="..." icon="..."` use case, as it doesn't require specifying any colors.